File: examples/nodejs/vimeo.js
const axios = require('axios');

const API_BASE = 'https://cog.api.br/api/v1';

/**
 * Download video from Vimeo
 * @param {string} url - Vimeo video URL
 */
async function downloadVimeo(url) {
  try {
    const response = await axios.get(`${API_BASE}/vimeo/download`, {
      params: { url }
    });

    if (response.data.success) {
      console.log('Vimeo Video Downloaded Successfully!');
      console.log('Title:', response.data.data.title);
      console.log('Author:', response.data.data.author);
      console.log('Quality:', response.data.data.quality);
      console.log('Views:', response.data.data.views);
      console.log('Likes:', response.data.data.likes);
      console.log('Download URL:', response.data.data.downloadUrl);
      console.log('Duration:', response.data.data.duration, 'seconds');
      console.log('Resolution:', `${response.data.data.width}x${response.data.data.height}`);
      
      return response.data.data;
    } else {
      console.error('Error:', response.data.error);
      return null;
    }
  } catch (error) {
    console.error('Request Error:', error.response?.data || error.message);
    return null;
  }
}

/**
 * Get available formats for Vimeo video
 * @param {string} url - Vimeo video URL
 */
async function getVimeoFormats(url) {
  try {
    const response = await axios.get(`${API_BASE}/vimeo/formats`, {
      params: { url }
    });

    if (response.data.success) {
      console.log('Available Formats:');
      response.data.formats.forEach((format, index) => {
        console.log(`\nFormat ${index + 1}:`);
        console.log('  ID:', format.formatId);
        console.log('  Quality:', format.quality);
        console.log('  Extension:', format.ext);
        console.log('  Size:', (format.filesize / 1024 / 1024).toFixed(2), 'MB');
      });
      
      return response.data.formats;
    } else {
      console.error('Error:', response.data.error);
      return null;
    }
  } catch (error) {
    console.error('Request Error:', error.response?.data || error.message);
    return null;
  }
}

/**
 * Get information about Vimeo video
 * @param {string} url - Vimeo video URL
 */
async function getVimeoInfo(url) {
  try {
    const response = await axios.get(`${API_BASE}/vimeo/info`, {
      params: { url }
    });

    if (response.data.success) {
      console.log('Vimeo Video Information:');
      console.log('Title:', response.data.info.title);
      console.log('Author:', response.data.info.author);
      console.log('Views:', response.data.info.views);
      console.log('Likes:', response.data.info.likes);
      console.log('Duration:', response.data.info.duration, 'seconds');
      console.log('Description:', response.data.info.description);
      console.log('Resolution:', `${response.data.info.width}x${response.data.info.height}`);
      
      return response.data.info;
    } else {
      console.error('Error:', response.data.error);
      return null;
    }
  } catch (error) {
    console.error('Request Error:', error.response?.data || error.message);
    return null;
  }
}

// Example usage
(async () => {
  console.log('=== Vimeo Download Example ===\n');
  
  // Example 1: Download Vimeo video
  const vimeoUrl = 'https://vimeo.com/123456789';
  console.log('Downloading from Vimeo:', vimeoUrl);
  await downloadVimeo(vimeoUrl);
  
  console.log('\n=== Vimeo Formats Example ===\n');
  
  // Example 2: Get available formats
  console.log('Getting formats for:', vimeoUrl);
  await getVimeoFormats(vimeoUrl);
  
  console.log('\n=== Vimeo Info Example ===\n');
  
  // Example 3: Get video info
  console.log('Getting video info:', vimeoUrl);
  await getVimeoInfo(vimeoUrl);
})();

module.exports = {
  downloadVimeo,
  getVimeoFormats,
  getVimeoInfo
};

File: examples/python/vimeo.py
import requests

API_BASE = 'https://cog.api.br/api/v1'

def download_vimeo(url: str) -> dict:
    """
    Download video from Vimeo
    
    Args:
        url: Vimeo video URL
        
    Returns:
        dict: Download data or None if error
    """
    try:
        response = requests.get(f'{API_BASE}/vimeo/download', params={'url': url})
        data = response.json()
        
        if data.get('success'):
            print('Vimeo Video Downloaded Successfully!')
            print(f"Title: {data['data']['title']}")
            print(f"Author: {data['data']['author']}")
            print(f"Quality: {data['data']['quality']}")
            print(f"Views: {data['data']['views']}")
            print(f"Likes: {data['data']['likes']}")
            print(f"Download URL: {data['data']['downloadUrl']}")
            print(f"Duration: {data['data']['duration']} seconds")
            print(f"Resolution: {data['data']['width']}x{data['data']['height']}")
            
            return data['data']
        else:
            print(f"Error: {data.get('error')}")
            return None
            
    except Exception as e:
        print(f'Request Error: {str(e)}')
        return None

def get_vimeo_formats(url: str) -> list:
    """
    Get available formats for Vimeo video
    
    Args:
        url: Vimeo video URL
        
    Returns:
        list: Available formats or None if error
    """
    try:
        response = requests.get(f'{API_BASE}/vimeo/formats', params={'url': url})
        data = response.json()
        
        if data.get('success'):
            print('Available Formats:')
            for index, format_info in enumerate(data['formats'], 1):
                print(f'\nFormat {index}:')
                print(f"  ID: {format_info['formatId']}")
                print(f"  Quality: {format_info['quality']}")
                print(f"  Extension: {format_info['ext']}")
                print(f"  Size: {format_info['filesize'] / 1024 / 1024:.2f} MB")
            
            return data['formats']
        else:
            print(f"Error: {data.get('error')}")
            return None
            
    except Exception as e:
        print(f'Request Error: {str(e)}')
        return None

def get_vimeo_info(url: str) -> dict:
    """
    Get information about Vimeo video
    
    Args:
        url: Vimeo video URL
        
    Returns:
        dict: Video information or None if error
    """
    try:
        response = requests.get(f'{API_BASE}/vimeo/info', params={'url': url})
        data = response.json()
        
        if data.get('success'):
            print('Vimeo Video Information:')
            print(f"Title: {data['info']['title']}")
            print(f"Author: {data['info']['author']}")
            print(f"Views: {data['info']['views']}")
            print(f"Likes: {data['info']['likes']}")
            print(f"Duration: {data['info']['duration']} seconds")
            print(f"Description: {data['info']['description']}")
            print(f"Resolution: {data['info']['width']}x{data['info']['height']}")
            
            return data['info']
        else:
            print(f"Error: {data.get('error')}")
            return None
            
    except Exception as e:
        print(f'Request Error: {str(e)}')
        return None

if __name__ == '__main__':
    print('=== Vimeo Download Example ===\n')
    
    # Example 1: Download Vimeo video
    vimeo_url = 'https://vimeo.com/123456789'
    print(f'Downloading from Vimeo: {vimeo_url}')
    download_vimeo(vimeo_url)
    
    print('\n=== Vimeo Formats Example ===\n')
    
    # Example 2: Get available formats
    print(f'Getting formats for: {vimeo_url}')
    get_vimeo_formats(vimeo_url)
    
    print('\n=== Vimeo Info Example ===\n')
    
    # Example 3: Get video info
    print(f'Getting video info: {vimeo_url}')
    get_vimeo_info(vimeo_url)
